src/app-layout/__tests__/drawers.test.tsx

```diff
@@ -0,0 +1,68 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React from 'react';
+import { act } from 'react-dom/test-utils';
+import { describeEachAppLayout, drawersConfigurations, renderComponent } from './utils';
+import AppLayout from '../../../lib/components/app-layout';
+import { useVisualRefresh } from '../../../lib/components/internal/hooks/use-visual-mode';
+import { useMobile } from '../../../lib/components/internal/hooks/use-mobile';
+import { KeyCode } from '../../../lib/components/internal/keycode';
+
+jest.mock('../../../lib/components/internal/hooks/use-mobile', () => ({
+  useMobile: jest.fn().mockReturnValue(true),
+}));
+
+describeEachAppLayout(() => {
+  test(`should not render drawer when it is not defined`, () => {
+    const { wrapper, rerender } = renderComponent(
+      <AppLayout contentType="form" {...drawersConfigurations.singleDrawer} />
+    );
+    expect(wrapper.findDrawersTriggers()).toHaveLength(1);
+    rerender(<AppLayout />);
+    expect(wrapper.findDrawersTriggers()).toHaveLength(0);
+  });
+});
+
+describe('Classic only features', () => {
+  beforeEach(() => {
+    (useVisualRefresh as jest.Mock).mockReturnValue(false);
+    (useMobile as jest.Mock).mockReturnValue(false);
+  });
+  afterEach(() => {
+    (useVisualRefresh as jest.Mock).mockReset();
+    (useMobile as jest.Mock).mockReset();
+  });
+
+  test(`Moves focus to slider when opened`, () => {
+    const { wrapper } = renderComponent(<AppLayout contentType="form" {...drawersConfigurations.resizableDrawer} />);
+
+    act(() => wrapper.findDrawersTriggers()[0].click());
+    expect(wrapper.findDrawersSlider()!.getElement()).toHaveFocus();
+  });
+
+  test('should change size in controlled mode', () => {
+    const onResize = jest.fn();
+    const drawersOpen = {
+      drawers: {
+        activeDrawerId: 'security',
+        onResize: (event: any) => onResize(event.detail),
+        items: drawersConfigurations.drawersResizableItems,
+      },
+    };
+    const { wrapper } = renderComponent(<AppLayout contentType="form" {...drawersOpen} />);
+    act(() => wrapper.findDrawersSlider().keydown(KeyCode.left));
+    expect(getComputedStyle(wrapper.findActiveDrawer().getElement()).width).toBe('290px');
+  });
```

Comment:
I'm not sure if I'm understanding this test correctly, but it looks like we're not actually controlling the size here since I don't see the `width` property (?)

I would recommend to also test that `onResize` has been called

Reply:
So without the `size` property, it will default to `280`. I wasn't sure how to test with adding a specific size, since if we explicitly set the size to a constant, `onResize` won't work. Is there a way in a testing environment to set the `size` property to something controlled by `onResize`?

Reply:
In a testing environment we don't have a state outside of the component, instead we have to replicate the updated state by rerendering the component with the new properties that we want to be passed in.

For example:
```js
const { wrapper, rerender } = renderComponent(
     <AppLayout contentType="form" {...drawersOpen} size={280} />
);
act(() => wrapper.findDrawersSlider().keydown(KeyCode.left));
expect(onResize).toHaveBeenCalledTimes(1);

// To check that the component is not using an uncontrolled state
expect(getComputedStyle(wrapper.findActiveDrawer().getElement()).width).toBe(280);


// Simulate that we have updated the controlled state to 320px
rerender(
     <AppLayout contentType="form" {...drawersOpen} size={320} />
);
expect(getComputedStyle(wrapper.findActiveDrawer().getElement()).width).toBe(320);

```